pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -407,6 +420,154 @@ func (c *Controller) setupIndex() *AmbientIndex {
 			}
 		},
 	}
+
+	idx.servicesMap = make(map[types.NamespacedName]*model.Service)
+	idx.handleServiceEntry = func(svc *model.Service, event model.Event) {
```

Comment:
Can this be an instance method?

Reply:
addressed in https://github.com/istio/istio/pull/45621/commits/707f66f0af47e5a8f1b8c4f986aa85a777d34030